src/Core/Prefab/TileBuilder.js

```diff
@@ -25,38 +25,47 @@ export default function newTileGeometry(builder, params) {
         let cachedBuffers = cacheBuffer.get(bufferKey);
         params.buildIndexAndUv_0 = !cachedBuffers;
         params.builder = builder;
-        return Promise.resolve(computeBuffers(params)).then((buffers) => {
-            if (!cachedBuffers) {
-                cachedBuffers = {};
-                cachedBuffers.index = new THREE.BufferAttribute(buffers.index, 1);
-                cachedBuffers.uv = new THREE.BufferAttribute(buffers.uvs[0], 2);
-
-                // Update cacheBuffer
-                cacheBuffer.set(bufferKey, cachedBuffers);
-            }
+        let buffers;
+        try {
+            buffers = computeBuffers(params);
+        } catch (e) {
+            return Promise.reject(e);
+        }
 
-            buffers.index = cachedBuffers.index;
-            buffers.uvs[0] = cachedBuffers.uv;
-            buffers.position = new THREE.BufferAttribute(buffers.position, 3);
-            buffers.normal = new THREE.BufferAttribute(buffers.normal, 3);
-            if (params.builder.uvCount > 1) {
-                buffers.uvs[1] = new THREE.BufferAttribute(buffers.uvs[1], 1);
-            }
+        if (!cachedBuffers) {
+            cachedBuffers = {};
+            cachedBuffers.index = new THREE.BufferAttribute(buffers.index, 1);
+            cachedBuffers.uv = new THREE.BufferAttribute(buffers.uvs[0], 2);
+
+            // Update cacheBuffer
+            cacheBuffer.set(bufferKey, cachedBuffers);
+        }
 
-            const geometry = new TileGeometry(params, buffers);
-            geometry.OBB = new OBB(geometry.boundingBox.min, geometry.boundingBox.max);
-
-            geometry._count = 0;
-            geometry.dispose = () => {
-                geometry._count--;
-                if (geometry._count == 0) {
-                    THREE.BufferGeometry.prototype.dispose.call(geometry);
-                    cacheTile.delete(south, params.level, bufferKey);
-                }
-            };
-            resolve(geometry);
-            return { geometry, quaternion, position };
-        });
+        buffers.index = cachedBuffers.index;
+        buffers.uvs[0] = cachedBuffers.uv;
+        buffers.position = new THREE.BufferAttribute(buffers.position, 3);
+        buffers.normal = new THREE.BufferAttribute(buffers.normal, 3);
+        if (params.builder.uvCount > 1) {
+            buffers.uvs[1] = new THREE.BufferAttribute(buffers.uvs[1], 1);
+        }
+
+        const geometry = new TileGeometry(params, buffers);
+        geometry.OBB = new OBB(geometry.boundingBox.min, geometry.boundingBox.max);
+
+        geometry._count = 0;
+        geometry.dispose = () => {
+            geometry._count--;
+            if (geometry._count <= 0) {
+                // to avoid remove index buffer and attribute buffer uv_0
+                // error un-bound buffer in webgl with VAO rendering
+                geometry.index = null;
+                delete geometry.attributes.uv_0;
```

Comment:
Could you leave a note for posterity, to change this, if there is a fix some times in THREE ?

Reply:
comment added